backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisSessionManager.java

```diff
@@ -7,97 +7,78 @@
 import org.joda.time.DateTime;
 import org.joda.time.Minutes;
 import org.molgenis.emx2.Database;
-import org.molgenis.emx2.MolgenisException;
 import org.molgenis.emx2.sql.SqlDatabase;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import spark.Request;
-import spark.Response;
 
 public class MolgenisSessionManager {
   public static final String MOLGENIS_TOKEN = "x-molgenis-token";
-  private static final String SESSION_ATTRIBUTE = "session";
+  private static final String SESSION_ATTRIBUTE = "molgenis_session";
   private static final Logger logger = LoggerFactory.getLogger(MolgenisSessionManager.class);
+  private static final int SESSION_TIMEOUT = 30;
 
-  // key is the user, might lead to trouble
+  // map so we can track the sessions
+  // session id is the key, todo in case of session less requests outside browser
   private Map<String, MolgenisSession> sessions = new LinkedHashMap<>();
```

Comment:
LinkedHashMap is not threadsafe!

Reply:
How will this lead to problems here?

Reply:
The inserts are done in a synchronized method but may run in parallel with reads done by clearAllCaches method which is not synchronized.

Reply:
changed to ConcurrentHashMap, will that help?